src/Lecture3.hs

```diff
@@ -82,7 +90,27 @@ Tuesday
   would work for **any** enumeration type in Haskell (e.g. 'Bool',
   'Ordering') and not just 'Weekday'?
 -}
-next = error "TODO"
+
+instance Enum Weekday where
+  fromEnum Monday    = 0
+  fromEnum Tuesday   = 1
+  fromEnum Wednesday = 2
+  fromEnum Thursday  = 3
+  fromEnum Friday    = 4
+  fromEnum Saturday  = 5
+  fromEnum Sunday    = 6
+
+  toEnum 0 = Monday
+  toEnum 1 = Tuesday
+  toEnum 2 = Wednesday
+  toEnum 3 = Thursday
+  toEnum 4 = Friday
+  toEnum 5 = Saturday
+  toEnum 6 = Sunday
+  toEnum a = toEnum (a `mod` 7)
```

Comment:
You're not supposed to implement this typeclass, it should be derived. And peculiarities of the `succ` behaviour should be handled inside the `next` function 👀